### PR TITLE
discover resources for override

### PR DESCRIPTION
**Pull request by AndrewNepogoda**
- Discover resource for data refresh.
- Bug fixes
- Delete data for resources marked for refresh.

### CODE REVIEW

File: cli-e2e-test/test_e2e.py [hunk 1]
```diff
@@ -21,21 +26,117 @@ class CliE2ETest(unittest.TestCase):
...
+                     "--end-date", "20230909"]
+        rsp = call(self.cmd_with_common_arguments + test_args + ["--drop-db"])
+        # then
+        self.assertNotEqual(rsp, 1)
+        # and when
+        rsp = call(self.cmd_with_common_arguments + test_args + ["--force-reimport"])
+        # then
+        self.assertNotEqual(rsp, 1)
+        rai_config = self.resource_manager.get_rai_config()
+        rps_json = workflow.rai.execute_relation_json(self.logger, rai_config, q.RESOURCES_TO_DELETE)
```
Comment (stanislav-bedunkevich) [inline cli-e2e-test/test_e2e.py:72]:
```suggestion
        rsp_json = workflow.rai.execute_relation_json(self.logger, rai_config, q.RESOURCES_TO_DELETE)
```

File: cli-e2e-test/test_e2e.py [hunk 2]
```diff
@@ -21,21 +26,117 @@ class CliE2ETest(unittest.TestCase):
...
+                     "--end-date", "20230909"]
+        rsp = call(self.cmd_with_common_arguments + test_args + ["--drop-db"])
+        # then
+        self.assertNotEqual(rsp, 1)
+        # and when
+        rsp = call(self.cmd_with_common_arguments + test_args)
+        # then
+        self.assertNotEqual(rsp, 1)
+        rai_config = self.resource_manager.get_rai_config()
+        rps_json = workflow.rai.execute_relation_json(self.logger, rai_config, q.RESOURCES_TO_DELETE)
```
Comment (stanislav-bedunkevich) [inline cli-e2e-test/test_e2e.py:56]:
```suggestion
        rsp_json = workflow.rai.execute_relation_json(self.logger, rai_config, q.RESOURCES_TO_DELETE)
```

File: workflow/query.py [hunk 3]
```diff
@@ -76,6 +76,107 @@ def insert:source_has_input_format(r,p) =
...
+        /*
+         * All simple sources are affected if they match with declared sources.
+         */
+        def potentially_affected_sources(rel, path) {{
+            source_declares_resource(rel, path) and
+            simple_sources(rel, path)
+        }}
+        /*
+         * All chunk partitioned sources for given date are affected in case new sources have at least one partition in this date.
+         */
```
Comment (stanislav-bedunkevich) [inline workflow/query.py:120]: could some part of this code go installed? with the flags being bounds and populated here, but the rest of the logic installed?

Comment (AndrewNepogoda) [reply workflow/query.py]: Moved

File: cli-e2e-test/test_e2e.py (hunk 1 above)
Comment (AndrewNepogoda) [reply cli-e2e-test/test_e2e.py]: fixed

File: cli-e2e-test/test_e2e.py (hunk 2 above)
Comment (AndrewNepogoda) [reply cli-e2e-test/test_e2e.py]: fixed

**Review by stanislav-bedunkevich — APPROVED** [review]
thanks!

**Pull request merged**